### PR TITLE
Issue#194

### LINKED ISSUE #194: Add onchain address validation compatible with Taproot adresses
**Is your feature request related to a problem? Please describe.**
Current on-chain payout (as on-the-fly swaps) does not accept taproot addressed. The RoboSats backend can pay to taproot. However, the function to validate an address submitted by the user does not support it.

**Describe the solution you'd like**
We are using pip package [coinaddrvalidator](https://github.com/nobitex/coinaddrvalidator) to validate on-chain addresses. Ideally, we would prefer to validate addresses using the LND backend, but this functionality does not seem to be implemented for gRPC calls (I think?). In addition, coinaddrvalidator is not well maintained and bloated with shitcoin address validators. It would be great to instead find a replacement or write our own on-chain address validation.

This is the relevant util that performs validation using coinaddrvalidator. The task consist in a) making it compatible with taproot and b) making it dependency free (or a light dependency).
https://github.com/Reckless-Satoshi/robosats/blob/aad87e7d98be30bbbb814b91534e0709724c2bc4/api/utils.py#L14-L43
Alternatively, a call to Bitcoin RPC or LND gRPC would also be good solutions.

**Additional context**
This task is **⚡rewarded with 120K Sats⚡** . Drop a comment if you want to be assigned.

### PR DESCRIPTION
**Pull request by zx9r**
Fixes #194 though not properly tested

### CODE REVIEW

File: .env-sample [hunk 1]
```diff
@@ -8,6 +8,11 @@ LND_CERT_BASE64='LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNLVENDQWRDZ0F3SUJBZ0l
 # base64 ~/.lnd/data/chain/bitcoin/testnet/admin.macaroon | tr -d '\n'
 LND_MACAROON_BASE64='AgEDbG5kAvgBAwoQsyI+PK+fyb7F2UyTeZ4seRIBMBoWCgdhZGRyZXNzEgRyZWFkEgV3cml0ZRoTCgRpbmZvEgRyZWFkEgV3cml0ZRoXCghpbnZvaWNlcxIEcmVhZBIFd3JpdGUaIQoIbWFjYXJvb24SCGdlbmVyYXRlEgRyZWFkEgV3cml0ZRoWCgdtZXNzYWdlEgRyZWFkEgV3cml0ZRoXCghvZmZjaGFpbhIEcmVhZBIFd3JpdGUaFgoHb25jaGFpbhIEcmVhZBIFd3JpdGUaFAoFcGVlcnMSBHJlYWQSBXdyaXRlGhgKBnNpZ25lchIIZ2VuZXJhdGUSBHJlYWQAAAYgMt90uD6v4truTadWCjlppoeJ4hZrL1SBb09Y+4WOiI0='
 
+# Bitcoin Core Daemon RPC, used to validate addresses
+BITCOIND_RPCURL = 'http://127.0.0.1:8332'
+BITCOIND_RPCUSER = 'rpc_user'
```
Comment (Reckless-Satoshi) [inline .env-sample:13]: Would you quickly fix the sample `BITCOIND_RPCUSER` & `BITCOIND_RPCUSER` to `robodev` so anyone running the dev stack does not have to care for these? These are set on `bitcoin.conf`
https://github.com/Reckless-Satoshi/robosats/blob/7645bb39f29ba55ec85d0e2b8caba8335eb57d04/docker/bitcoind/bitcoin.conf#L8-L9

**Comment by zx9r** [comment]
I modified it in my fork/branch but I dont know how to bring it here to this pull request. Should I create another pull request or is it automatically updated ?

File: .env-sample (hunk 1 above)
Comment (Reckless-Satoshi) [inline .env-sample]: Would you quickly fix the sample `BITCOIND_RPCUSER` & `BITCOIND_RPCUSER` to `robodev` so anyone running the dev stack does not have to care for these? These are set on `bitcoin.conf`
https://github.com/Reckless-Satoshi/robosats/blob/7645bb39f29ba55ec85d0e2b8caba8335eb57d04/docker/bitcoind/bitcoin.conf#L8-L9

File: .env-sample [hunk 2]
```diff
@@ -8,6 +8,11 @@ LND_CERT_BASE64='LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNLVENDQWRDZ0F3SUJBZ0l
 # base64 ~/.lnd/data/chain/bitcoin/testnet/admin.macaroon | tr -d '\n'
 LND_MACAROON_BASE64='AgEDbG5kAvgBAwoQsyI+PK+fyb7F2UyTeZ4seRIBMBoWCgdhZGRyZXNzEgRyZWFkEgV3cml0ZRoTCgRpbmZvEgRyZWFkEgV3cml0ZRoXCghpbnZvaWNlcxIEcmVhZBIFd3JpdGUaIQoIbWFjYXJvb24SCGdlbmVyYXRlEgRyZWFkEgV3cml0ZRoWCgdtZXNzYWdlEgRyZWFkEgV3cml0ZRoXCghvZmZjaGFpbhIEcmVhZBIFd3JpdGUaFgoHb25jaGFpbhIEcmVhZBIFd3JpdGUaFAoFcGVlcnMSBHJlYWQSBXdyaXRlGhgKBnNpZ25lchIIZ2VuZXJhdGUSBHJlYWQAAAYgMt90uD6v4truTadWCjlppoeJ4hZrL1SBb09Y+4WOiI0='
 
+# Bitcoin Core Daemon RPC, used to validate addresses
+BITCOIND_RPCURL = 'http://127.0.0.1:8332'
```
Comment (Reckless-Satoshi) [inline .env-sample:12]: Tested! And it works like a charm. 
Needs a quick fix here to be seamless with the development environment. The default RPC port for testnet is 18332.

It rejects mainnet addresses in testnet as expected "Invalid bench32 format". Unable to connect to RPC shows the right error message. And made up strings response is "Invalid format".

**Comment by Reckless-Satoshi** [comment]
Hey @zx9r,

Ready to merge, nice work!

As described in https://github.com/Reckless-Satoshi/robosats/issues/194#issuecomment-1200405769 we have 2 concurrent PRs for the same rewarded task, we will be merging this fix instead of #200 .

Given that both solutions are valid, let's split reward of #194 60/40 for the two submitted PRs. Please submit an LN invoice (from a proxy wallet or in private) for 72K Sats

**Comment by zx9r** [comment]
Thank you !
Here is the invoice: lnbc720u1p3wdy33pp5d7293wdvv98efnfyzwz2tu0a43756r6sgcrmglcldfqwva67724qdqu2askcmr9wssx7e3q2dshgmmndp5scqzpgxqyz5vqsp5zdjsthapznt3q08fz3ycegra3cjudc4rnutpwsldgvc44rhknvdq9qyyssqqecrzc9qdxxw6zv2vkgxeftsrz232pk3y9pcxj59cyhrvzruugs948lch396jy4ree3wqs52xx5n73t97pc649wrc5ljt03raj7qy0sq7dj7sv

**Comment by Reckless-Satoshi** [comment]
40e02d7c4706f023a6279d8b4473633ecc5b0611c9dae1e5c8208abaf701452e

**Pull request merged**